### PR TITLE
Core/Spells: Implement SPELL_EFFECT_PLAY_SCENE

### PR DESCRIPTION
**Pull request by Golrag**
**Changes proposed:**

-  Implement SPELL_EFFECT_PLAY_SCENE

**Target branch(es):**

- master

**Tests performed:**
Tested in game with spell 193709:
- Added scene_template for scene 1109
```sql
INSERT INTO `scene_template` (`SceneId`, `Flags`, `ScriptPackageID`, `ScriptName`) VALUES (1109, 11, 1489, '');

```
- .cast 193709

**Known issues and TODO list:**

- [ ] These spells have basepoints for the play_scene effect, no idea what they are used for: 130374, 130375, 130376, 130377, 134013, 134023, 134039, 134210, 134574, 135799

### CODE REVIEW

File: src/server/game/Spells/SpellEffects.cpp [hunk 1]
```diff
@@ -266,7 +266,7 @@ pEffect SpellEffects[TOTAL_SPELL_EFFECTS]=
     &Spell::EffectNULL,                                     //195 SPELL_EFFECT_195
     &Spell::EffectNULL,                                     //196 SPELL_EFFECT_196
     &Spell::EffectNULL,                                     //197 SPELL_EFFECT_197
-    &Spell::EffectNULL,                                     //198 SPELL_EFFECT_PLAY_SCENE
+    &Spell::EffectPlayScene,                                //198 SPELL_EFFECT_PLAY_SCENE
```
Comment (joschiwald) [inline src/server/game/Spells/SpellEffects.cpp]: you forgot to update the comment in SpellEffectName enum in SharedDefines.h

**Comment by joschiwald** [comment]
@Golrag pls update UsedTargetObjectType in SpellEffectInfo::_data too

File: src/server/game/Spells/SpellInfo.cpp [hunk 2]
```diff
@@ -920,7 +920,7 @@ SpellEffectInfo::StaticData SpellEffectInfo::_data[TOTAL_SPELL_EFFECTS] =
     {EFFECT_IMPLICIT_TARGET_NONE,     TARGET_OBJECT_TYPE_NONE}, // 195 SPELL_EFFECT_195
     {EFFECT_IMPLICIT_TARGET_NONE,     TARGET_OBJECT_TYPE_NONE}, // 196 SPELL_EFFECT_196
     {EFFECT_IMPLICIT_TARGET_NONE,     TARGET_OBJECT_TYPE_NONE}, // 197 SPELL_EFFECT_197
-    {EFFECT_IMPLICIT_TARGET_NONE,     TARGET_OBJECT_TYPE_NONE}, // 198 SPELL_EFFECT_PLAY_SCENE
+    {EFFECT_IMPLICIT_TARGET_CASTER,   TARGET_OBJECT_TYPE_UNIT_AND_DEST}, // 198 SPELL_EFFECT_PLAY_SCENE
```
Comment (joschiwald) [inline src/server/game/Spells/SpellInfo.cpp]: none, dest should be right here

Comment (Golrag) [inline src/server/game/Spells/SpellInfo.cpp]: So 
{EFFECT_IMPLICIT_TARGET_NONE,     TARGET_OBJECT_TYPE_UNIT_AND_DEST}
or
{EFFECT_IMPLICIT_TARGET_NONE,     TARGET_OBJECT_TYPE_DEST}

I havent really looked much into it and just made sure it still worked in game :/

**Pull request merged**